internal/credential/attributes.go

```diff
@@ -58,11 +72,32 @@ func GetCredentialsConfig(in *structpb.Struct, region string) (*awsutil.Credenti
 		return nil, errors.InvalidArgumentError("Error in the secrets provided", badFields)
 	}
 
-	return awsutil.NewCredentialsConfig(
-		awsutil.WithAccessKey(accessKey),
-		awsutil.WithSecretKey(secretKey),
-		awsutil.WithRegion(region),
-	)
+	opts := []awsutil.Option{
+		awsutil.WithEnvironmentCredentials(true),
+	}
+	if accessKey != "" && secretKey != "" {
```

Comment:
I wonder if we should explicitly error if there is a conflicting configuration, for example these and RoleArn are provided. Is there a scenario that both are valid.

Reply:
That's a good point, I'm not sure why the user would provide both credential types. If they did, I believe that the static credentials would still be used. 

For example, the user can provide static credentials that are incorrectly configured and ec2 role provider credentials that work as expected. Although we have both, awsutil will still prioritize the static credentials and disregard using the ec2 role provider credentials.